app/models/timeline/step_duplication.rb

```diff
@@ -99,9 +88,16 @@ def apply_time_interval(date, frequency, time_unit, month_repeat_on)
         reference_date = date.beginning_of_month + interval - 1.month
         first_try = Chronic.parse("#{month_repeat_on} of next month", now: reference_date)
 
-        # Sometimes 5th weekday of month doesn't exist
-        if first_try.nil? && month_repeat_on =~ /\A5th/
-          Chronic.parse("#{month_repeat_on.sub('5th', '4th')} of next month", now: reference_date)
+        if first_try.nil?
+          # Sometimes 5th weekday of month doesn't exist
+          if month_repeat_on =~ /\A5th/
+            Chronic.parse("#{month_repeat_on.sub('5th', '4th')} of next month", now: reference_date)
+          # Error if day to repeat doesn't exist in some months
+          elsif month_repeat_on.to_i > 28
+            (date + interval).end_of_month
+          else
+            raise "Error parsing date string for monthly repeat"
+          end
```

Comment:
Can you please update the specs to reveal the previous flaw here? Any time you fix a bug in a model like this it's a good idea to cover the change with a spec. Even better is to do it TDD style.